File: php/features.php
<?php
declare(strict_types=1);

// DataDragon SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class DataDragonFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new DataDragonBaseFeature();
            case "test":
                return new DataDragonTestFeature();
            default:
                return new DataDragonBaseFeature();
        }
    }
}

File: py/datadragon_sdk/features.py
# DataDragon SDK feature factory

from datadragon_sdk.feature.base_feature import DataDragonBaseFeature
from datadragon_sdk.feature.test_feature import DataDragonTestFeature


def _make_feature(name):
    features = {
        "base": lambda: DataDragonBaseFeature(),
        "test": lambda: DataDragonTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
